src/ast/analysis/AstTypeEnvironmentAnalysis.cpp

```diff
@@ -14,13 +14,13 @@
  *
  ***********************************************************************/
 
-#include "AstTypeEnvironmentAnalysis.h"
-#include "AstAttribute.h"
-#include "AstProgram.h"
-#include "AstTranslationUnit.h"
-#include "AstType.h"
+#include "ast/analysis/AstTypeEnvironmentAnalysis.h"
 #include "GraphUtils.h"
-#include "TypeSystem.h"
+#include "ast/AstAttribute.h"
```

Comment:
What's the include ordering rule being used?

Reply:
Lexicographical. So here it's capitals, then lower case. `main.cpp` is a more comprehensive example.

Since we use lower case for subdirectories/namespaces, it prioritises 'local' includes.